src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -692,6 +661,46 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
         })
     }
 
+    fn resolve_variant_or_field(
+        &self,
+        item_name: Symbol,
+        did: DefId,
+        ty_res: Res,
+        extra_fragment: &Option<String>,
+    ) -> Option<Result<(Res, Option<String>), ErrorKind<'path>>> {
+        debug!("looking for variants or fields named {} for {:?}", item_name, did);
+        // FIXME(jynelson): why is this different from
+        // `variant_field`?
+        match self.cx.tcx.type_of(did).kind() {
+            ty::Adt(def, _) => {
```

Comment:
nit: I would make this an early return so you don't have so much indentation.
```suggestion
        let def = match self.cx.tcx.type_of(did).kind() {
            ty::Adt(def, _) => def,
            _ => return None,
        };
```